### PR TITLE
Use rAF in tests by default

### PR DESCRIPTION
**Pull request by brendenpalmer**

### CODE REVIEW

File: tests/unit/scheduler-test.js [hunk 1]
```diff
@@ -136,4 +154,40 @@ module('Unit | Scheduler', function(hooks) {
 
     assert.equal(_getScheduleFn(), SIMPLE_CALLBACK);
   });
+
+  test('route idle and route painted schedule functions are correct', function(assert) {
+    assert.expect(2);
+
+    assert.equal(_getWhenRouteIdleScheduleFn(), requestIdleCallback);
```
Comment (rwjblue) [inline tests/unit/scheduler-test.js]: This seems somewhat brittle to me, I would prefer not to test the private APIs and rely on their coverage by way of the public APIs that we expose...

Comment (brendenpalmer) [reply tests/unit/scheduler-test.js]: Agreed, good call out — I've updated tests

**Comment by scalvert** [comment]
@brendenpalmer does your IDE reorder imports and exports? There's a lot of unnecessary changes in this PR that make it really hard to distinguish what's needed and what's cosmetic.

Can you revert the import/export changes?

**Comment by brendenpalmer** [comment]
@scalvert Alright, just reverted those changes

**Comment by scalvert** [comment]
@rwjblue Brenden discovered an issue, which we haven't confirmed yet, where the test waiter never completes in tests when we're using `whenRouteIdle`. The hypothesis is that during tests, the main thread is pinned and never gets into an idle state, thus resulting in the `requestIdleCallback` to never fire.

As I mentioned, it's not confirmed, but he was able to resolve this by switching to using `requestAnimationFrame` within tests only. This PR is attempting to allow for configurability during testing. 

I'd like to do two things:
1. Confirm this is in fact the issue before we bake that functionality into this addon
2. Revisit the proposed changes, as they feel like we're adding a fair bit of complexity into something that was quite straightforward.

@rwjblue happy to hear your thoughts.

**Comment by scalvert** [comment]
Closing in favor of #280

**Pull request closed without merging**